### PR TITLE
Add angular and styles package

### PR DESCRIPTION
**Pull request by cal-smith**
~Waiting on #1983~

This PR:

- adds the angular package
- pulls the styles out of the react package and:
  - adds `@ai-apps/styles` as a package
  - adds that package as a dependency for Angular
  - **does not** make it a dependency for React. We'll migrate react at a later date, for now nothing changes for React, and the process we use to extract the styles from the React package is easily repeatable.
- adds build processes for both new packages
- vendors all style dependencies 
- properly enables yarn workspaces for the monorepo (should result in better package de-deuplication among other things)
- adds a non-published `@ai-apps/monorepo-utils` (naming open for discussion!) as a place to collect shared tooling
  - includes a tool to handle vendoring packages

I'll add more here as I go, but this PR is up as a draft for early visibility.

### CODE REVIEW

File: packages/angular/.editorconfig
Comment (davidicus) [inline packages/angular/.editorconfig:6]: is this purposely different from the root editor config?

File: packages/angular/.storybook/preview.js
Comment (davidicus) [inline packages/angular/.storybook/preview.js:12]: Are the two packages not going to use the same theme? Should this be G10?

Comment (cal-smith) [reply packages/angular/.storybook/preview.js:12]: Good catch ... We use the G100 theme for the storybook UI in `carbon-components-angular`, but I'll switch it over to G10 to match the React package

File: packages/angular/.editorconfig
Comment (cal-smith) [reply packages/angular/.editorconfig:6]: It was ... Given that I've formatted everything with Prettier now I'm guessing they'll be in sync, and this is unnecessary

File: packages/angular/.storybook/preview.js
Comment (davidicus) [inline packages/angular/.storybook/preview.js:12]: still doing the G100 here. Should change to G10.

File: packages/angular/.storybook/theme.js
Comment (davidicus) [inline packages/angular/.storybook/theme.js:6]: should change to G10 to avoid confusion

**Review by davidicus — APPROVED** [review]
This seems to look good. Just a few comments about the theme. I can not really speak to the angular bits.

File: packages/angular/.storybook/preview.js
Comment (cal-smith) [reply packages/angular/.storybook/preview.js:12]: 🤦

**Comment by cal-smith** [comment]
> This seems to look good. Just a few comments about the theme. I can not really speak to the angular bits.

Cool 👍 I know @zvonimirfras was working on a review for the Angular specific bits

**Review by zvonimirfras — COMMENTED** [review]
From when there was more files 😛 
I'll do another round on recent changes.

File: .editorconfig
Comment (cal-smith) [reply .editorconfig:8]: Existing PAL convention is 2 spaces (enforced by prettier) so that's what everything is going with

File: packages/angular/src/toolkit/page-header/page-header.component.ts
Comment (cal-smith) [reply packages/angular/src/toolkit/page-header/page-header.component.ts]: The AI Apps one is a bit different, so I'm going to leave this one alone for now ... Once we have a better idea if we can merge theme seamlessly we could do a rename

File: packages/angular/src/index.stories.ts
Comment (cal-smith) [reply packages/angular/src/index.stories.ts]: That's prettier ... It's setup to run globally, to keep consistency within the PAL ... A bit weird coming from CCA, but I figure we kinda have to adopting the existing conventions in this case

File: packages/angular/src/toolkit/page-header/page-header.component.ts
Comment (cal-smith) [reply packages/angular/src/toolkit/page-header/page-header.component.ts]: Same as the prior comment ... I've also already updated any of the toolkit components that should be updated so I'm just going to resolve these suggestions

File: packages/angular/src/toolkit/sortable-list/sortable-list-item.component.ts
Comment (cal-smith) [reply packages/angular/src/toolkit/sortable-list/sortable-list-item.component.ts:1]: I'm going to try adding this one and see what happens 🤔 I'm not sure if this is prettiered already or not

**Review by zvonimirfras — CHANGES_REQUESTED** [review]

File: packages/angular/custom-webpack.config.js
Comment (cal-smith) [reply packages/angular/custom-webpack.config.js:5]: That's prettier I believe ... I'm not a fan of it either, and I believe it's configurable, but I don't know if that's a change everyone is on board with. I'm not sure what the best way to get consensus on a change like that would be, opening an issue and posting to slack? (@davidicus if you have any thoughts here ...)

File: packages/angular/src/button/button.directive.spec.ts
Comment (cal-smith) [reply packages/angular/src/button/button.directive.spec.ts:3]: yes?

File: packages/angular/src/button/button.directive.ts
Comment (cal-smith) [reply packages/angular/src/button/button.directive.ts:21]: See my other comments

File: packages/angular/tslint.json
Comment (cal-smith) [reply packages/angular/tslint.json:17]: See my other comments - we're aligning to existing repo standards

File: packages/angular/src/toolkit/draggable/draggable.directive.ts
Comment (cal-smith) [reply packages/angular/src/toolkit/draggable/draggable.directive.ts:1]: I tried this before and it blew everything up. At this point all formatting is handled by prettier (and a couple lint rules that probably could be tidied up a bit)

File: packages/angular/src/toolkit/table-settings/table-settings.stories.ts
Comment (cal-smith) [reply packages/angular/src/toolkit/table-settings/table-settings.stories.ts:11]: I don't see how this would be a useful change in this case ... I kinda like the firewalling we've given the `toolkit/` directory with the `sc` namespace - keeps all that weirdness isolated

File: packages/angular/src/toolkit/draggable/droppable.directive.ts
Comment (cal-smith) [reply packages/angular/src/toolkit/draggable/droppable.directive.ts:1]: Not changing

File: packages/angular/tslint.json
Comment (cal-smith) [reply packages/angular/tslint.json:89]: See other comments on this topic

File: packages/angular/src/toolkit/sortable-list/sortable-list.component.ts
Comment (cal-smith) [reply packages/angular/src/toolkit/sortable-list/sortable-list.component.ts:13]: See the warning comment in the component. I'm not going to change this one over since its _very_ non-compliant

File: packages/angular/src/toolkit/sortable-list/sortable-list.component.ts
Comment (cal-smith) [reply packages/angular/src/toolkit/sortable-list/sortable-list.component.ts:18]: I've provided comments before ... I guess it hid them when I clicked "resolve"?

File: packages/angular/src/toolkit/table/head/sterling-table-head-cell.component.ts
Comment (cal-smith) [reply packages/angular/src/toolkit/table/head/sterling-table-head-cell.component.ts:6]: The aiTable is going to be a bit of a beast, we'll need to do a proper evaluation to see what we can carry over

File: packages/angular/src/toolkit/table/head/sterling-table-head.component.ts
Comment (cal-smith) [reply packages/angular/src/toolkit/table/head/sterling-table-head.component.ts:16]: See other table comment

File: packages/angular/tslint.json
Comment (cal-smith) [reply packages/angular/tslint.json:29]: I'll re-format this one 👍

File: packages/angular/src/toolkit/draggable/draggable.directive.ts
Comment (zvonimirfras) [reply packages/angular/src/toolkit/draggable/draggable.directive.ts:1]: _sigh_

**Review by zvonimirfras — APPROVED** [review]

**Comment by cal-smith** [comment]
🚨 Don't merge _yet_ we need to finish setting up the ai-apps npm scope and making sure our token has enough permissions to publish everything 🚨

**Pull request merged**
